File: Algorithm/boj/그리디/11501 주식.py
T = int(input());

for _ in range(T):
    n = int(input())
    stock = list(map(int,input().split()))
    answer = 0 
    maxStock = stock[-1]
    for i in range(n-2,-1,-1):
        if stock[i] > maxStock: 
            maxStock = stock[i]
        else:
            answer += maxStock-stock[i]
    print(answer)

File: Algorithm/boj/그리디/11501 주식.js
function solution(stock) {
  let result = 0;
  let maxStock = stock[stock.length - 1];
  for (let i = stock.length - 2; i >= 0; i--) {
    if (stock[i] < maxStock) {
      result += maxStock - stock[i];
    } else {
      maxStock = stock[i];
    }
  }
  console.log(result);
}

const fs = require("fs");
const input = fs.readFileSync("dev/stdin").toString().trim().split("\n");
const T = Number(input[0]);

for (let i = 0; i < T; i++) {
  const N = Number(input[i]);
  let stock = [];
  for (let j = 0; j < N; j++) {
    stock[j] = Number(input[j]);
  }
  solution(stock);
}
